server/graphql/v2/mutation/AccountMutations.ts

```diff
@@ -48,6 +48,47 @@ const accountMutations = {
       });
     },
   },
+  addAccountTwoFactorAuth: {
+    type: new GraphQLNonNull(Account),
+    description: 'Add 2FA to the Account if it does not have it',
+    args: {
+      account: {
+        type: new GraphQLNonNull(AccountReferenceInput),
+        description: 'Account that will have 2FA added to it',
+      },
+      secret: {
+        type: new GraphQLNonNull(GraphQLString),
+        description: 'The generated secret to save to the Account',
+      },
+    },
+    async resolve(_, args, req): Promise<object> {
+      if (!req.remoteUser) {
+        throw new Unauthorized();
+      }
+
+      return sequelize.transaction(async transaction => {
```

Comment:
I don't see a need for a transaction here, there's no concurrency issue. If we decide to use one, we need to pass it to all the queries (`User.findOne`, `User.update`).